File: examples/python/search_subagent_queries.py
from __future__ import annotations

import json
from pathlib import Path


REPO_ROOT = Path(__file__).resolve().parents[2]
CATALOG_PATH = REPO_ROOT / "data" / "subagent-query-catalog.json"


def main() -> None:
    catalog = json.loads(CATALOG_PATH.read_text())
    matches = [
        entry for entry in catalog["entries"]
        if entry["domain"] == "security" and entry["mode"] == "harden"
    ]

    for entry in matches[:5]:
        print(f'{entry["id"]}: {entry["title"]}')


if __name__ == "__main__":
    main()

File: examples/javascript/search-subagent-queries.mjs
import fs from "node:fs";
import path from "node:path";
import { fileURLToPath } from "node:url";

const __filename = fileURLToPath(import.meta.url);
const __dirname = path.dirname(__filename);
const repoRoot = path.resolve(__dirname, "..", "..");
const catalogPath = path.join(repoRoot, "data", "subagent-query-catalog.json");

const catalog = JSON.parse(fs.readFileSync(catalogPath, "utf8"));
const matches = catalog.entries.filter(
  (entry) => entry.domain === "security" && entry.mode === "harden"
);

for (const entry of matches.slice(0, 5)) {
  console.log(`${entry.id}: ${entry.title}`);
}
